core/common/buffer_back_insert_iterator.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common/buffer.hpp"
+
+namespace std {
+
+/*
+ * std::back_insert_iterator is an output iterator
+ * that appends to a container for which it was constructed.
+ */
+template <>
+class back_insert_iterator<kagome::common::Buffer> {
+public:
+  using value_type = std::decay_t <decltype(std::declval<kagome::common::Buffer&>().operator[](0))>;
```

Comment:
just add `using value_type = uint8_t` to Buffer and use it here

this looks very complicated